ddtrace/tracer/spancontext.go

```diff
@@ -390,63 +394,72 @@ func (t *trace) finishedOne(s *span) {
 		// all the spans in the trace, so the below conditions will not
 		// be accurate and would trigger a pre-mature flush, exposing us
 		// to a race condition where spans can be modified while flushing.
+		//
+		// TODO(partialFlush): should we do a partial flush in this scenario?
```

Comment:
Hmm I would say when partial flushing is enabled by default it makes sense, but the only way to get here today would be if the partial flushing limit is configured larger than the max payload limit (10000). Maybe we should instead log a warning if that configuration is set since it means partial flushing will never happen?

Reply:
True. Or perhaps in addition to logging a warning we can just cap the max for partial flushing to be the max payload limit, and if someone gives something higher than that, we just lower it to 10000. I'll move the TODO here and we can handle it in https://github.com/DataDog/dd-trace-go/pull/2056 instead.